File: aio/social-distancing.py
distin = open("distin.txt", "r")
distout = open("distout.txt", "w")
 
N, K = list(map(int, distin.readline().split()))
 
locs = []
 
for a in range(N):
    locs.append(int(distin.readline()))
 
if N == 2:
    if abs(locs[0] - locs[1]) >= K:
        nums = 2
    else:
        nums = 1
 
elif K == 1:
    locs = set(locs)
    nums = len(locs)
 
else:
    locs.sort()
    nums = 0
    for n in range(N):
        if n == 0:
            dist = 0
            nums += 1
        else:
            dist += locs[n] - locs[n-1]
            if dist >= K:
                nums += 1
                dist = 0

distout.write(str(nums))

File: aio/social-distancing.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>

int main() {
    int n, k;
    int startzero = 0; int startone = 0;

     freopen("distin.txt", "r", stdin);
     freopen("distout.txt", "w", stdout);

    std::cin >> n >> k;

    int locs[n];
    int last;

    for (int i = 0; i < n; i++) {
        std::cin >> locs[i];
    }

    std::sort(locs, locs+n);

    for (int i=0; i < n; i++) {
        if (i == 0) {
            startzero ++;
            last = locs[i];
        }

        else {
            if (locs[i] - last >= k) {
                startzero ++;
                last = locs[i];
            }
        }
    }

    std::cout << std::max(startzero, startone);

    return 0;
}
